api/v1beta2/network_types.go

```diff
@@ -94,12 +104,57 @@ var (
 
 // TargetGroupHealthCheck defines health check settings for the target group.
 type TargetGroupHealthCheck struct {
-	Protocol        *string `json:"protocol,omitempty"`
-	Path            *string `json:"path,omitempty"`
-	Port            *string `json:"port,omitempty"`
-	IntervalSeconds *int64  `json:"intervalSeconds,omitempty"`
-	TimeoutSeconds  *int64  `json:"timeoutSeconds,omitempty"`
-	ThresholdCount  *int64  `json:"thresholdCount,omitempty"`
+	// The protocol to use to health check connect with the target.
+	// Defaults to TCP.
+	// +kubebuilder:default=TCP
+	// +kubebuilder:validation:Enum=TCP;HTTP;HTTPS
+	// +optional
+	Protocol *string `json:"protocol,omitempty"`
+
+	// The destination for health checks on the targets when using the protocol HTTP or HTTPS,
+	// otherwise the path will be ignored.
+	// Defaults to "/readyz".
+	// +kubebuilder:default="/readyz"
+	// +optional
+	Path *string `json:"path,omitempty"`
+
+	// The port the load balancer uses when performing health checks on targets.
+	// Defaults to "6443".
+	// +kubebuilder:default="6443"
+	// +optional
+	Port *string `json:"port,omitempty"`
+
+	// The approximate amount of time, in seconds, between health checks of an individual
+	// target.
+	// Defaults to 10.
+	// +kubebuilder:default=10
+	// +kubebuilder:validation:Minimum=10
+	// +optional
+	IntervalSeconds *int64 `json:"intervalSeconds,omitempty"`
+
+	// The amount of time, in seconds, during which no response from a target means
+	// a failed health check.
+	// Defaults to 5.
+	// +kubebuilder:default=5
+	// +kubebuilder:validation:Minimum=5
+	// +optional
+	TimeoutSeconds *int64 `json:"timeoutSeconds,omitempty"`
+
+	// The number of consecutive health check successes required before considering
+	// a target healthy.
+	// Defaults to 5.
```

Comment:
Just to clarify, are these the _current_ defaults? What's the current behavior if left unset?

Reply:
Yes, the current default for CLB. Those values comes from the default static set to CLB[1], that I am proposing to moving it to constants[2] to be reused in ELBv2.
[1] https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4846/files#diff-ee38bda5efe2606e7e7746a556c49a2d53d14eca60c029b8efebea657b8718f1L1005-L1008
[2] https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4846/files#diff-1c519200705fb576b64e7fb271f7f683a2e6efadd54abd2a70b243e5069490e9R32-R39

Reply:
@vincepri for your later question, thanks for poping it up. I am revising this current fix. The goal here is to propose a fix in the API's TG health check setting it correctly, and not change or expose the API. So I removed the updates in `TargetGroupHealthCheck`, setting the defaults using the same method as CLB.

The PR #4849 is addressing the wider scope, proposing changes in the API allowing users to customize the health checks setting it accordingly when creating LB, addressing it as a feature instead of a bug fix.

[1] https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4849/files#diff-ee38bda5efe2606e7e7746a556c49a2d53d14eca60c029b8efebea657b8718f1

Let me know what do you think, you all, @vincepri @nrb @faiq .

Reply:
I left my comments on that PR.

For this one, I think it is fine. From my reading this PR is simply setting the same health check for ELBv2 as there currently is for ELBv1. Can you please confirm if I got that right?

Reply:
> this PR is simply setting the same health check for ELBv2 as there currently is for ELBv1
> Can you please confirm if I got that right?

@faiq yes.